test/infrastructure/docker/internal/docker/machine.go

```diff
@@ -173,18 +173,22 @@ func (m *Machine) ProviderID() string {
 	return fmt.Sprintf("docker:////%s", m.ContainerName())
 }
 
-// Address will get the IP address of the machine. If IPv6 is enabled, it will return
-// the IPv6 address, otherwise an IPv4 address.
-func (m *Machine) Address(ctx context.Context) (string, error) {
+// Address will get the IP address of the machine. It can return
+// a single IPv4 address, a single IPv6 address or one of each depending on the machine.ipFamily.
+func (m *Machine) Address(ctx context.Context) ([]string, error) {
 	ipv4, ipv6, err := m.container.IP(ctx)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-
-	if m.ipFamily == clusterv1.IPv6IPFamily {
-		return ipv6, nil
+	switch m.ipFamily {
+	case clusterv1.IPv6IPFamily:
+		return []string{ipv6}, nil
+	case clusterv1.IPv4IPFamily:
+		return []string{ipv4}, nil
+	case clusterv1.DualStackIPFamily:
+		return []string{ipv6, ipv4}, nil
```

Comment:
nit. Should we return ipv4 first in this case? (just because of a general tendency to default to ipv4 first)